File: backend/app/schemas/audit.py
"""
审核相关 Schema
定义回复审核列表、详情、统计等请求/响应数据结构
"""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class AuditItemResponse(BaseModel):
    """审核项响应"""
    id: UUID
    review_id: UUID
    store_name: Optional[str] = None
    store_id: Optional[str] = None
    user_name: Optional[str] = None
    user_avatar: Optional[str] = None
    rating: Optional[int] = None
    content: Optional[str] = None
    platform: Optional[str] = None
    ai_reply: Optional[str] = None
    status: str
    risk_level: Optional[str] = None
    scores: Optional[dict] = None
    reject_reason: Optional[str] = None
    auditor_name: Optional[str] = None
    reviewed_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    model_config = {"from_attributes": True}


class AuditListResponse(BaseModel):
    """审核列表响应"""
    items: list[AuditItemResponse]
    total: int
    page: int
    pageSize: int


class AuditStatsResponse(BaseModel):
    """审核统计响应"""
    pending_count: int = 0
    approved_count: int = 0
    rejected_count: int = 0
    sent_count: int = 0
    total_count: int = 0
    avg_processing_time: Optional[float] = None


class AuditApproveRequest(BaseModel):
    """审核通过请求"""
    pass


class AuditRejectRequest(BaseModel):
    """审核拒绝请求"""
    reason: str = Field(..., description="拒绝原因", min_length=1)


class AuditRegenerateRequest(BaseModel):
    """重新生成回复请求"""
    pass


class AuditActionResponse(BaseModel):
    """审核操作响应"""
    id: UUID
    status: str
    ai_reply: Optional[str] = None
    reviewed_at: Optional[datetime] = None
    auditor_name: Optional[str] = None
    model_config = {"from_attributes": True}

File: src/api/audit.ts
import { api } from '@/lib/api';

export interface AuditItem {
  id: string;
  review_id: string;
  store_name: string | null;
  store_id?: string;
  user_name: string | null;
  user_avatar?: string | null;
  rating: number | null;
  content: string | null;
  platform: string | null;
  ai_reply: string | null;
  status: 'pending' | 'approved' | 'rejected' | 'sent';
  risk_level: string | null;
  scores: {
    realism: number;
    empathy: number;
    concreteness: number;
    consistency: number;
  } | null;
  reject_reason?: string | null;
  auditor_name?: string | null;
  reviewed_at?: string | null;
  created_at: string | null;
}

export interface AuditStats {
  pending_count: number;
  approved_count: number;
  rejected_count: number;
  sent_count: number;
  total_count: number;
  avg_processing_time: number | null;
}

export const auditApi = {
  getAuditList: async (params?: {
    status?: string;
    keyword?: string;
    page?: number;
    limit?: number;
  }): Promise<{ items: AuditItem[]; total: number; page: number; pageSize: number }> => {
    const searchParams = new URLSearchParams();
    if (params) {
      Object.entries(params).forEach(([key, value]) => {
        if (value !== undefined && value !== null && value !== '') {
          searchParams.append(key, String(value));
        }
      });
    }
    const response = await api.get<any>(`/v1/audit/list?${searchParams.toString()}`);
    return response.data || response;
  },

  getAuditById: async (id: string): Promise<AuditItem> => {
    const response = await api.get<any>(`/v1/audit/${id}`);
    return response.data || response;
  },

  approveAudit: async (id: string): Promise<any> => {
    const response = await api.post(`/v1/audit/${id}/approve`, {});
    return response.data || response;
  },

  rejectAudit: async (id: string, reason: string): Promise<any> => {
    const response = await api.post(`/v1/audit/${id}/reject`, { reason });
    return response.data || response;
  },

  regenerateReply: async (id: string): Promise<any> => {
    const response = await api.post(`/v1/audit/${id}/regenerate`, {});
    return response.data || response;
  },

  getStats: async (): Promise<AuditStats> => {
    const response = await api.get<any>('/v1/audit/stats');
    return response.data || response;
  },
};
